### PR TITLE
Sort dragged files so images are written first, then js, then css, then html for BlobUtils caching

### PR DESCRIPTION
**Pull request by humphd**
I forgot to think about CSS also using image files when I did the sort order on dragged files.  We have to write them to the fs in the right order, or external resources won't be found when needed.  This fixes it.

### CODE REVIEW

File: src/utils/DragAndDrop.js [hunk 1]
```diff
@@ -181,12 +181,29 @@ define(function (require, exports, module) {
...
+                // prior to rewriting an HTML file.
+                function rateFileByType(filename) {
+                    var ext = Path.extname(filename);
+
+                    // We want to end up with: [images, ..., js, ..., css, html]
+                    // since CSS can include images, and HTML can include CSS or JS.
+                    // We also treat .md like an HTML file, since we render them.
+                    if(Content.isHTML(ext) || Content.isMarkdown(ext)) {
+                        return 10;
+                    } else if(Content.isCSS(ext)) {
```
Comment (gideonthomas) [inline src/utils/DragAndDrop.js]: Do we need this check? Even if CSS has a priority of 3, we would still end up with [images, css, js, html] wouldn't we?

Comment (sedge) [inline src/utils/DragAndDrop.js]: I guess the question is, do we care if CSS comes before JS?

**Comment by sedge** [comment]
r+, w/ minor nit

File: src/utils/DragAndDrop.js (hunk 1 above)
Comment (humphd) [inline src/utils/DragAndDrop.js]: We do no-rewriting of URLs in JavaScript files, so they can come anywhere.  Same for images, JSON files, etc.  We rewrite CSS files to include images, so the images have to be written first, or won't exist.  We rewrite HTML to fix CSS, JS, and images, so they have to come last.

I think what I have is right.

**Pull request merged**